File: 1465-maximum-area-of-a-piece-of-cake-after-horizontal-and-vertical-cuts/1465-maximum-area-of-a-piece-of-cake-after-horizontal-and-vertical-cuts.py
class Solution(object):
    def maxArea(self, h, w, hCs, vCs):
        """
        :type h: int
        :type w: int
        :type horizontalCuts: List[int]
        :type verticalCuts: List[int]
        :rtype: int
        """
        hCs.sort()
        vCs.sort()
        
        hCs = hCs + [h]
        vCs = vCs + [w]
        
        maxH = hCs[0]
        maxV = vCs[0]
        
        for i in xrange(1, len(hCs)):
            n = hCs[i] - hCs[i - 1]
            if (n > maxH): maxH = n
        
        for i in xrange(1, len(vCs)):
            n = vCs[i] - vCs[i - 1]
            if (n > maxV): maxV = n
        
        return (maxH*maxV)%(10**9 + 7)

File: 1465-maximum-area-of-a-piece-of-cake-after-horizontal-and-vertical-cuts/1465-maximum-area-of-a-piece-of-cake-after-horizontal-and-vertical-cuts.c
int mulMod(int a, int b, int m) {
    int res = 0;
    a %= m;
    b %= m;
    while (b > 0) {
        if (b%2 == 1) res = (res + a)%m;
        a = (a*2)%m;
        b = b/2;
    }
    return res;
}


int comp(const void* a, const void* b) {
    return *((int*)a) - *((int*)b);
}


int maxArea(int h, int w, int* hC, int hCS, int* vC, int vCS){
    
    qsort(hC, hCS, sizeof(int), comp);
    qsort(vC, vCS, sizeof(int), comp);
    
    int maxV = w - vC[vCS - 1];
    int maxH = h - hC[hCS - 1];
    
    if (hC[0] > maxH) maxH = hC[0];
    if (vC[0] > maxV) maxV = vC[0];
    
    int i;
    for(i = 1; i < hCS; i++) {
        if (hC[i] - hC[i - 1] > maxH) maxH = hC[i] - hC[i - 1];
    }
    
    for(i = 1; i < vCS; i++) {
        if (vC[i] - vC[i - 1] > maxV) maxV = vC[i] - vC[i - 1];
    }
    
    
    return mulMod(maxH, maxV, 1000000000 + 7);

}